### PR TITLE
svelted up movesList component and refactured class Move and BaseMove…

### PR DESCRIPTION
**Pull request by oodtoon**
… also with a hint of TS error fix
READ THE CHARTS!

### CODE REVIEW

File: src/lib/components/MoveList.svelte [hunk 1]
```diff
@@ -1,71 +1,91 @@
...
   };
+
 </script>
 
 <div class="moves-list">
     <h3>moves list</h3>
-    <ol />
+    <ol>
+      {#each $moveList as move, i }
+      {@debug move}
```
Comment (dgore7) [inline src/lib/components/MoveList.svelte]: remove the debug

File: src/lib/components/MoveList.svelte [hunk 2]
```diff
@@ -1,71 +1,91 @@
...
+  //     this.listRoot?.appendChild(this.currentListItem);
+  //   }
+
+  //   setResult(result) {
+  //     this.gameResult = document.createElement("div");
+  //     this.gameResult.classList.add("result");
+  //     this.gameResult.textContent = result;
+  //     this.listRoot?.append(this.gameResult);
+  //   }
+  // }
```
Comment (dgore7) [inline src/lib/components/MoveList.svelte]: should be deleted

File: src/lib/components/Game.svelte [hunk 3]
```diff
@@ -132,11 +136,8 @@
     updatePlayerTurnAndText(prevPlayer);
   }
 
-  function updateMovesList(event) {
-    if (movesList.currentListItem.children.length === 2) {
-      movesList.nextListItem();
-    }
-    movesList.addMove(event.detail.move.toString());
+  function updateMovesList(event: MouseEvent) {
```
Comment (dgore7) [inline src/lib/components/Game.svelte]: move would be a more natural parameter here.
```suggestion
  function updateMovesList(move: Move) {
```

File: src/lib/models/board.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ type PieceConstructor = {
 
 export default class Board {
   static LANE_SIZE = 8;
-  #board: Piece[][] = [];
+  #board: (Piece | null)[][] = [];
```
Comment (dgore7) [inline src/lib/models/board.ts]: good candidate for extraction

Comment (dgore7) [reply src/lib/models/board.ts]:
```suggestion
  type BoardEntry = Piece | null
  type BoardGrid = BoardEntry[][]
  ... 
  #board: BoardGrid = [];
```

Comment (oodtoon) [reply src/lib/models/board.ts]: Just want to discuss this first so leaving comment

**Pull request merged**